### PR TITLE
Introduce :time subproject

### PR DESCRIPTION
**Pull request by lptr**
This is our monotonic clock implementation, and a few related timer utilities. The logic for this comes from `:base-services`'s `org.gradle.internal.time` package.

Also moved to the new subproject: `ExponentialBackoff`.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

File: platforms/core-runtime/time/build.gradle.kts [hunk 1]
```diff
@@ -14,22 +14,15 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.io;
-
-import org.gradle.internal.InternalTransformer;
-import org.gradle.internal.concurrent.CompositeStoppable;
-
-import java.io.Closeable;
+plugins {
```
Comment (asodja) [inline platforms/core-runtime/time/build.gradle.kts:17]: 🤯 What happened with this diff?

**Review by asodja — APPROVED** [review]
LGTM

**Pull request merged**